diff --git a/ptenum.py b/ptenum.py
index e638b8c..d079232 100644
--- a/ptenum.py
+++ b/ptenum.py
@@ -63,7 +63,7 @@ from __future__ import annotations
 
 import struct, logging, textwrap
 from builtins import object
-from past.utils import old_div
+from operator import floordiv as old_div
 from typing import Dict, Tuple, Generator, List, Type, Optional
 from functools import wraps, cached_property, lru_cache, cache
 
