diff --git a/anydesk.py b/anydesk.py
index d7af54a..56b5e6d 100644
--- a/anydesk.py
+++ b/anydesk.py
@@ -17,7 +17,7 @@ class AnyDesk(interfaces.plugins.PluginInterface, timeliner.TimeLinerInterface):
     def get_requirements(cls):
         return [requirements.ModuleRequirement(name='kernel', description='Windows kernel',
                                                architectures=["Intel32", "Intel64"]),
-                requirements.PluginRequirement(name='filescan', plugin=filescan.FileScan, version=(0, 0, 0))]
+                requirements.VersionRequirement(name='filescan', component=filescan.FileScan, version=(2, 0, 0))]
 
     @classmethod
     def parse_trace_file(cls, file_raw, file_name):
@@ -88,7 +88,7 @@ class AnyDesk(interfaces.plugins.PluginInterface, timeliner.TimeLinerInterface):
     def generate_timeline(self):
         kernel = self.context.modules[self.config['kernel']]
         for row in self._generator(
-                filescan.FileScan.scan_files(self.context, kernel.layer_name, kernel.symbol_table_name)):
+                filescan.FileScan.scan_files(self.context, self.config['kernel'])):
             _depth, row_data = row
             description = "{} : {} ".format(
                 row_data[1], row_data[4])
@@ -102,4 +102,4 @@ class AnyDesk(interfaces.plugins.PluginInterface, timeliner.TimeLinerInterface):
             ("Time", datetime.datetime),
             ("Context", str),
             ("Message", str)],
-            self._generator(filescan.FileScan.scan_files(self.context, kernel.layer_name, kernel.symbol_table_name)))
+            self._generator(filescan.FileScan.scan_files(self.context, self.config['kernel'])))
diff --git a/inodes.py b/inodes.py
index 7fcd865..7998076 100644
--- a/inodes.py
+++ b/inodes.py
@@ -21,7 +21,7 @@ class Inodes(plugins.PluginInterface, timeliner.TimeLinerInterface):
         return [
             requirements.ModuleRequirement(name='kernel', description='Linux kernel',
                                            architectures=["Intel32", "Intel64"]),
-            requirements.PluginRequirement(name='pslist', plugin=pslist.PsList, version=(2, 0, 0)),
+            requirements.VersionRequirement(name='pslist', component=pslist.PsList, version=(4, 0, 0)),
             requirements.VersionRequirement(name='linuxutils', component=linux.LinuxUtilities, version=(2, 0, 0)),
             requirements.ListRequirement(name='pid',
                                          description='Filter on specific process IDs',
diff --git a/keepass.py b/keepass.py
index 4623ca1..8267404 100644
--- a/keepass.py
+++ b/keepass.py
@@ -23,8 +23,8 @@ class KeePass(interfaces.plugins.PluginInterface):
                 description="Windows kernel",
                 architectures=["Intel32", "Intel64"],
             ),
-            requirements.PluginRequirement(
-                name="pslist", plugin=pslist.PsList, version=(2, 0, 0)
+            requirements.VersionRequirement(
+                name="pslist", component=pslist.PsList, version=(3, 0, 0)
             ),
             requirements.IntRequirement(
                 name="pid",
@@ -118,8 +118,7 @@ class KeePass(interfaces.plugins.PluginInterface):
             self._generator(
                 pslist.PsList.list_processes(
                     context=self.context,
-                    layer_name=kernel.layer_name,
-                    symbol_table=kernel.symbol_table_name,
+                    kernel_module_name=self.config["kernel"],
                     filter_func=filter_func,
                 )
             ),
diff --git a/prefetch.py b/prefetch.py
index 58a9581..9ffae4f 100644
--- a/prefetch.py
+++ b/prefetch.py
@@ -236,7 +236,7 @@ class Prefetch(interfaces.plugins.PluginInterface):
     def get_requirements(cls):
         return [requirements.ModuleRequirement(name='kernel', description='Windows kernel',
                                                architectures=["Intel32", "Intel64"]),
-                requirements.PluginRequirement(name='filescan', plugin=filescan.FileScan, version=(0, 0, 0)), ]
+                requirements.VersionRequirement(name='filescan', component=filescan.FileScan, version=(2, 0, 0)), ]
 
     @classmethod
     def version_17(cls, prefetch_file):
@@ -468,4 +468,4 @@ class Prefetch(interfaces.plugins.PluginInterface):
             ("FileSize", int),
             ("PrefetchHash", format_hints.Hex),
             ("LastExecution", datetime.datetime), ("ExecutionCounter", int)],
-            self._generator(filescan.FileScan.scan_files(self.context, kernel.layer_name, kernel.symbol_table_name)))
+            self._generator(filescan.FileScan.scan_files(self.context, self.config['kernel'])))
